tests/tests/test-pov/test-evm-over-pov.ts

```diff
@@ -0,0 +1,133 @@
+    .createType("Compact<u32>", `0x${BigInt((evmCode.length + 1) * 2).toString(16)}`)
+    .toHex(true)}${evmCode}`;
+
+  // Create the batchs of contracts to deploy
+  const batchs = contracts
+    .reduce(
+      (acc, value) => {
+        if (acc[acc.length - 1].length >= 30) acc.push([]);
+        if (!value.deployed) acc[acc.length - 1].push([value.key, storageData]);
+        return acc;
+      },
+      [[]] as [string, string][][]
+    )
+    .filter((batch) => batch.length > 0);
+
+  // Set the storage of the contracts
+  let nonce = await context.web3.eth.getTransactionCount(alith.address);
+  for (let i = 0; i < batchs.length; i++) {
+    const batch = batchs[i];
+    await context.createBlock([
+      context.polkadotApi.tx.sudo
+        .sudo(context.polkadotApi.tx.system.setStorage(batch))
+        .signAsync(alith, {
+          nonce: nonce++,
+        }),
+    ]);
+  }
+  return contracts as HeavyContract[];
+};
+
+describeDevMoonbeam("PoV Limit (4.2Mb in Dev)", (context) => {
+  let contractProxy: Contract;
+  let contracts: HeavyContract[];
+
+  before("Deploy the contracts", async function () {
+    // Deploy the CallForwarder contract
+    const creation = await createContract(context, "CallForwarder");
+    contractProxy = creation.contract;
+    await context.createBlock(creation.rawTx);
+
+    // Deploy heavy contracts (test won't use more than what is needed for reaching max pov)
+    contracts = await deployHeavyContracts(
+      context,
+      6000,
+      6000 + Math.ceil(MAX_BLOCK_DEV_POV / 24_000) + 1
+    );
+  });
+
+  it("should allow to produce block just under the PoV Limit", async function () {
+    const max_contracts = Math.floor(MAX_BLOCK_DEV_POV / 24_000) - 1;
+
+    const { result, block } = await context.createBlock(
+      createTransaction(context, {
+        to: contractProxy.options.address,
+        data: contractProxy.methods
+          .callRange(contracts[0].account, contracts[max_contracts].account)
+          .encodeABI(),
+      })
+    );
+
+    debug("block.proof_size", block.proof_size);
+    expect(block.proof_size).to.be.at.least(MAX_BLOCK_DEV_POV - 20_000);
+    expect(block.proof_size).to.be.at.most(MAX_BLOCK_DEV_POV - 1);
+    // The transaction should be not be included in the block
+    expect(result.successful).to.equal(true);
+  });
+
+  it("should prevent a transaction reaching just over the PoV", async function () {
+    const max_contracts = Math.ceil(MAX_BLOCK_DEV_POV / 24_000);
+
+    const { result, block } = await context.createBlock(
+      createTransaction(context, {
+        to: contractProxy.options.address,
+        data: contractProxy.methods
+          .callRange(contracts[0].account, contracts[max_contracts].account)
+          .encodeABI(),
+      })
+    );
+
+    debug("block.proof_size", block.proof_size);
+    // Empty blocks usually do not exceed 10kb, picking 50kb as a safe limit
+    expect(block.proof_size).to.be.at.most(50_000);
```

Comment:
The txn included above is expected not to be included, right? Should this test show that it wasn't included or at least that the block was empty?